### PR TITLE
[42][37] Implement signin page and logout button functionality

### PR DESCRIPTION
**Pull request by miello**
## What did you do

- Implement Signin page
- Implement Logout functionality
- Fix missing page guard

## Demo

https://dev.modela.miello.dev/login

image

### CODE REVIEW

File: apps/web/src/modules/job/detail/pages/index.test.tsx [hunk 1]
```diff
@@ -8,6 +8,14 @@ describe('<JobDetailPage />', () => {
...
   }))
   const useRouter = jest.spyOn(require('next/router'), 'useRouter')
+  jest.doMock('common/context/UserContext', () => ({
+    useUser: () => ({
+      user: {
+        isVerified: true,
+        type: UserType.ACTOR,
+      },
+    }),
+  }))
```
Comment (PKhing) [inline apps/web/src/modules/job/detail/pages/index.test.tsx:18]: Is `mockUser()` enough 
ref: https://pkhing.notion.site/Front-Mock-useUser-bde9aa10d8eb4747b242ce4b2009bafe

**Review by PKhing — CHANGES_REQUESTED** [review]
Thai text for this
image
ฝากเพิ่ม icon login ที
image

File: apps/web/src/modules/job/detail/pages/index.test.tsx (hunk 1 above)
Comment (miello) [reply apps/web/src/modules/job/detail/pages/index.test.tsx:18]: Well yes, I have already forgotten this utility function 😂

File: apps/web/src/modules/login/hooks/useLoginForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        await apiClient.post<unknown, unknown, LoginDto>('/auth/login', data)
+        await refetch()
+        router.push('/job')
+      } catch (err) {
+        const error = err as AxiosError<{ message: string }>
+
+        if (error.response?.status === 401)
+          displaySnackbar('อีเมลหรือรหัสผ่านไม่ถูกต้อง', 'error')
+
+        displaySnackbar('เกิดข้อผิดพลาดที่ไม่ทราบสาเหตุ', 'error')
```
Comment (PKhing) [inline apps/web/src/modules/login/hooks/useLoginForm/index.tsx:39]: 😂

**Review by PKhing — CHANGES_REQUESTED** [review]

**Review by PKhing — APPROVED** [review]

**Pull request merged**